File: week08/19/P1.py
def solution(K, A):
    count = 0
    temp = 0
    for i in A:
        if (temp+i < K):
            temp += i
        else:
            count += 1
            temp = 0
    return count

if __name__ == "__main__":
    A = [1,2,3,4,1,1,3]
    # A = [2]
    print(solution(4, A))

File: week08/19/P1.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    int solution(int K, vector<int> &A) {
        int count = 0, temp = 0;
        for (int i=0; i<A.size(); i++) {
            if (temp+A[i] < K) temp += A[i];
            else {
                count++;
                temp = 0;
            }
        }
        return count;
    }
};

int main() {
    vector<int> A = {1,2,3,4,1,1,3};
    Solution result;
    cout << result.solution(4, A) << endl;

    return 0;
}